src/oisst/Geometry/Geometry.cc

```diff
@@ -9,42 +9,59 @@
 
 #include "eckit/config/Configuration.h"
 
+#include "atlas/grid.h"
+
 #include "oops/util/abor1_cpp.h"
 
 namespace oisst {
 
 // ----------------------------------------------------------------------------
 
-  Geometry::Geometry(const eckit::Configuration & conf,
-                     const eckit::mpi::Comm & comm)
-    : comm_(comm) {
-    util::abor1_cpp("Geometry::Geometry() needs to be implemented.",
-                    __FILE__, __LINE__);
+  Geometry::Geometry(const eckit::Configuration & conf, const eckit::mpi::Comm & comm) : comm_(comm) {
+  //util::abor1_cpp("Geometry::Geometry() needs to be implemented.", __FILE__, __LINE__);
+    
+    atlas::RegularLonLatGrid atlasRllGrid("Slat360x180");
+    atlasFunctionSpace_.reset(new atlas::functionspace::StructuredColumns(atlasRllGrid, 
+                                  atlas::option::halo(0)));
+    
+    atlasFieldSet_.reset(new atlas::FieldSet());
   }
 
 // ----------------------------------------------------------------------------
 
-  Geometry::Geometry(const Geometry & other)
-    : comm_(other.comm_) {
-    util::abor1_cpp("Geometry::Geometry() needs to be implemented.",
-                     __FILE__, __LINE__);
+  Geometry::Geometry(const Geometry & other) : comm_(other.comm_) {
+  //util::abor1_cpp("Geometry::Geometry() needs to be implemented.", __FILE__, __LINE__);
+
+    atlasFunctionSpace_.reset(new atlas::functionspace::StructuredColumns(other.atlasFunctionSpace_->grid(), 
```

Comment:
I'm not sure if all this is needed in the copy constructor here, it might work if the unique pointers were changed to shared pointers, and you just copy the pointers instead. Doesn't really matter at this point though, we can worry about that later when it's time to optimize the code

Reply:
Yes let's worry about it later, right now what I (and Jim) want is to get an initial result first. Later on we will optimize the code as I get more familiar with JEDI and ATLAS etc.